x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java

```diff
@@ -359,15 +361,15 @@ protected NodeTermsEnumResponse dataNodeOperation(NodeTermsEnumRequest request,
                         request.searchAfter()
                     );
                     if (terms != null) {
-                        shardTermsEnums.add(terms);
+                        shardTermsEnums.add(new ShardTermsEnum(terms, mappedFieldType::valueForDisplay));
```

Comment:
We still need to maintain the connection from a shard level enumeration to the field type that generated it since we can ask for terms enumerations across indices that have mixed type mappings for the same field name (i.e. keyword for older indices, version for newer ones).